### DISCUSSION TITLE
GPL:: Saving JPEG?

### DISCUSSION THREAD

**Discussion by ApeachM** [The-OpenROAD-Project/OpenROAD#3822] [Q&A] [answered]
Hello. I want to use how to save the JEPG files during debugging `GPL`.

[In this code](https://github.com/ApeachM/OpenROAD/blob/c6bd7c29293fd47a23212384267517b213f77293/src/gpl/src/nesterovPlace.cpp#L427C8-L438), there seems to exist the method to make some images during `GPL`. 

```tcl
read_lef ispd18/ispd18_test1/ispd18_test1.input.lef
read_def ispd18/ispd18_test1/ispd18_test1.input.def
gpl::global_placement_debug
global_placement -skip_initial_place
```

I did `gpl::global_placement_debug` before calling `global_placement`. 

I can see the snapshots during `GPL` in `GUI`, but I don't know how to save them automatically for each pause step. How do I do it?

Also, I want to color for a specific cell, editing c++ code in `gpl` for my debugging. Is this possible?

**Comment by ApeachM** [comment]
I solved it with `gui::Gui::get()->saveImage(file_name);`. Thank you.

**Reply by maliberty** [reply, marked as answer]
I guess that must be in c++.  From tcl you can use save_image.  You can look at Graphics::drawObjects for the drawing code.
